File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        p=strs[0]
        n=len(p)
        for i in strs[1:]:
            while p!=i[0:n]:
                n=n-1
                if n == 0:
                    return ""
                p=p[0:n]
        return p

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string s=strs[0];
        int n=s.size();
        for (int i=1; i<strs.size();i++){
            while (s!=strs[i].substr(0,n)){
                n=n-1;
                if(n==0){
                    return "";
                }
                s=s.substr(0,n);
            }
        }
        return s;
    }
};